backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/ApplicationCachePerUser.java

```diff
@@ -0,0 +1,166 @@
+package org.molgenis.emx2.web;
+
+import static org.molgenis.emx2.Constants.ANONYMOUS;
+import static org.molgenis.emx2.web.MolgenisWebservice.oidcController;
+import static org.pac4j.core.util.Pac4jConstants.USERNAME;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import graphql.GraphQL;
+import io.javalin.http.Context;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpSession;
+import java.time.Duration;
+import org.molgenis.emx2.Database;
+import org.molgenis.emx2.DatabaseListener;
+import org.molgenis.emx2.MolgenisException;
+import org.molgenis.emx2.Schema;
+import org.molgenis.emx2.graphql.GraphqlApiFactory;
+import org.molgenis.emx2.sql.JWTgenerator;
+import org.molgenis.emx2.sql.SqlDatabase;
+import org.molgenis.emx2.tasks.ScriptTableListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ApplicationCachePerUser {
```

Comment:
we could instead design this as:

applicationCache.getUserCache(ctx).getDatabase()
applicationCache.getUserCache(ctx).getSchema(schemaName)
applicationCache.getUserCache(ctx).getDatabaseGraphql()
applicationCache.getUserCache(ctx).getSchemaGraphql(schemaName)

what do you think?